### PR TITLE
MEP_oM: Addition of Generic Mechanical Equipment object

### LINKED ISSUE #1094: MEP_oM: Addition of Generic Mechanical Equipment object
#### Description:
Addition of generic Mechanical Equipment object, properties to include:
-*New* Enum of MechanicalEquipmentType
-Parts
-Location/Point

### PR DESCRIPTION
**Pull request by kayleighhoude**
### Issues addressed by this PR

Closes #1094 

Generic Mechanical Equipment Object added
MechanicalEquipmentType enum added
image

Workflow Prototype (assign type enum after pull):
image

### Additional comments
@Hmullerjones to review the enum for the mechanical equipment types for UK understanding of my American verbiage :)

### CODE REVIEW

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check code compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check code compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check code compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check code compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check code compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

File: MEP_oM/Enums/MechanicalEquipmentType.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using System.ComponentModel;
```
Comment (FraserGreenroyd) [inline MEP_oM/Enums/MechanicalEquipmentType.cs:1]: Copyright header needed

File: MEP_oM/Equipment/MechanicalEquipment.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.MEP.Equipment.Parts;
+using BH.oM.MEP.Enums;
+using BH.oM.Geometry;
+
+namespace BH.oM.MEP.Equipment
+{
+    class MechanicalEquipment
```
Comment (FraserGreenroyd) [inline MEP_oM/Equipment/MechanicalEquipment.cs:9]: Class definition should be `public` and inherit from `BHoMObject` - `public class MechanicalEquipment : BHoMObject`

File: MEP_oM/Equipment/MechanicalEquipment.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Generic;
```
Comment (FraserGreenroyd) [inline MEP_oM/Equipment/MechanicalEquipment.cs:1]: Copyright header needed

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Some substantial compliance changes needed for this PR please @kayleighhoude .

Also, this PR risks conflict with https://github.com/BHoM/BHoM/pull/1089 - whichever is ready first will get to merge first, and the other will have to be resolved around it.

Finally, a reminder on [branch naming conventions](https://github.com/BHoM/documentation/wiki/Resolving-an-Issue#branch-naming-conventions) for future reference please, we don't want to fall into bad habits :smile:

File: MEP_oM/Enums/MechanicalEquipmentType.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        Undefined,
+        AirHandlingUnit,
+        AirSeparator,
+        AirSourceHeatPump,
+        Boiler,
+        CabinetUnitHeater,
+        Chiller,
+        CoolingTower,
+        ExpansionTankVessel,
```
Comment (Hmullerjones) [inline MEP_oM/Enums/MechanicalEquipmentType.cs:18]: You don't necessarily need the word vessel here - as long as the word expansion is there, but it doesn't hurt

Comment (Hmullerjones) [reply MEP_oM/Enums/MechanicalEquipmentType.cs:18]: Everything else looks good

**Review by Hmullerjones — APPROVED** [review]

File: MEP_oM/Enums/MechanicalEquipmentType.cs (hunk 1 above)
Comment (kayleighhoude) [reply MEP_oM/Enums/MechanicalEquipmentType.cs]: Done in latest commit :)

File: MEP_oM/Equipment/MechanicalEquipment.cs (hunk 3 above)
Comment (kayleighhoude) [reply MEP_oM/Equipment/MechanicalEquipment.cs]: Done in latest commit :)

File: MEP_oM/Equipment/MechanicalEquipment.cs (hunk 2 above)
Comment (kayleighhoude) [reply MEP_oM/Equipment/MechanicalEquipment.cs]: Done in latest commit :)

**Review by FraserGreenroyd — APPROVED** [review]
LGTM now

**Pull request merged**